src/libcore/tests/iter.rs

```diff
@@ -291,6 +307,27 @@ fn test_zip_nth_side_effects() {
     assert_eq!(b, vec![200, 300, 400, 500, 600]);
 }
 
+#[test]
+fn test_zip_nth_back_side_effects() {
+    let mut a = Vec::new();
+    let mut b = Vec::new();
+    let value = [1, 2, 3, 4, 5, 6]
+        .iter()
+        .cloned()
+        .map(|n| {
+            a.push(n);
+            n * 10
+        })
+        .zip([2, 3, 4, 5, 6, 7, 8].iter().cloned().map(|n| {
+            b.push(n * 100);
+            n * 1000
+        }))
+        .nth_back(3);
+    assert_eq!(value, Some((30, 4000)));
+    assert_eq!(a, vec![6, 6, 5, 5, 4, 4, 3]);
```

Comment:
This is not correct and appears to be a bug that exists in the current `next_back` implementation. This should only contain each number once.

Reply:
Hm, so the current `nth_back` implementation is also wrong?
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=ad07ad6a7077e2ba38eb232eed5e15bc

Reply:
It's calling `next_back`, so yes.

Reply:
Filed it as #68536 (I think this PR encounters another roadblock that Niko pointed out, so left a new issue).